### PR TITLE
Add nbytes as a dependency and reuse code from Node.js

### PR DESCRIPTION
**Pull request by anonrig**
> Blocked by https://github.com/nodejs/nbytes/pull/3

Several weeks ago, @jasnell and I moved `nbytes` into a different repository to reuse Node.js' implementation. Here's the follow up of that work.

Nbytes: https://github.com/nodejs/nbytes

### CODE REVIEW

File: src/workerd/api/node/buffer.c++ [hunk 1]
```diff
@@ -171,7 +101,7 @@ uint32_t writeInto(
       // Fall-through
     case Encoding::BASE64URL: {
       auto str = kj::str(string);
-      return base64_decode(
+      return nbytes::Base64Decode(
```
Comment (jasnell) [inline src/workerd/api/node/buffer.c++:104]: potentially open for switching to simdutf?

**Review by jasnell — APPROVED** [review]

File: src/workerd/api/node/buffer.c++ (hunk 1 above)
Comment (anonrig) [reply src/workerd/api/node/buffer.c++:104]: yup, I added a TODO for decoding, but handle them all in a follow up PR

**Comment by jasnell** [comment]
lgtm once CI passes and the internal PR is open adding the dependency there

File: src/workerd/api/node/buffer.c++ [hunk 2]
```diff
@@ -564,11 +495,21 @@ jsg::Optional<uint32_t> BufferUtil::indexOf(
 void BufferUtil::swap(jsg::Lock& js, kj::Array<kj::byte> buffer, int size) {
   if (buffer.size() <= 1) return;
   switch (size) {
-    case 16: return SwapBytes<uint16_t>(buffer);
-    case 32: return SwapBytes<uint32_t>(buffer);
-    case 64: return SwapBytes<uint64_t>(buffer);
+  case 16: {
+    JSG_REQUIRE(nbytes::SwapBytes16(buffer.asChars().begin(), buffer.size()), Error, "Swap bytes failed")
```
Comment (jasnell) [inline src/workerd/api/node/buffer.c++:499]:
```suggestion
    JSG_REQUIRE(nbytes::SwapBytes16(buffer.asChars().begin(), buffer.size()), Error, "Swap bytes failed");
```

Comment (anonrig) [reply src/workerd/api/node/buffer.c++]: ...

**Review by mikea — APPROVED** [review]

**Pull request merged**